packages/contracts/contracts/foundry/SphinxUtils.sol

```diff
@@ -536,8 +536,34 @@ contract SphinxUtils is SphinxConstants, StdUtils {
             _config.saltNonce == 0 &&
             bytes(_config.orgId).length == 0
         ) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    function fetchSphinxConfig(address _script) public returns (SphinxConfig memory) {
+        ISphinxScript(_script).configureSphinx();
+        return ISphinxScript(_script).sphinxFetchConfig();
+    }
+
+    function fetchAndValidateConfig(address _script) public returns (SphinxConfig memory) {
+        SphinxConfig memory config = fetchSphinxConfig(_script);
+        validate(config);
+        return config;
+    }
+
+    /**
+     * @notice Performs validation on the user's deployment. This mainly checks that the user's
+     *         configuration is valid. This validation occurs regardless of the `SphinxMode` (e.g.
+     *         proposals, broadcasting, etc).
+     */
+    function validate(SphinxConfig memory _config) public {
+        // We still explicitly check if the config is empty b/c you could define the sphinxConfig
+        // function, but not actually configure any options in it.
+        if (isConfigObjectEmpty(_config)) {
             revert(
-                "Sphinx: Detected an empty 'sphinxConfig' struct. Did you forget to add fields to it in your script's\nsetUp function or constructor? If you've already added fields to it in your setUp function, have you\ncalled 'super.setUp()' in any contracts that inherit from your script?"
```

Comment:
`vm.expectRevert()` only works on the immediate next function call. If you do a call within that call before reverting, then `vm.expectRevert()` won't catch the revert, even if it is correct. So, to keep the `validate` function easy to test, I left it still accepting a `SphinxConfig` object and then created a new function, `fetchAndValidateConfig`, which fetches, validates, and returns the config object. So we'll usually want to use that function. It's not required, though; it's just generally more convenient, IMO.